sopel/modules/admin.py

```diff
@@ -247,61 +247,85 @@ def mode(bot, trigger):
     bot.write(('MODE', bot.nick + ' ' + mode))
 
 
-@sopel.module.require_privmsg("This command only works as a private message.")
-@sopel.module.require_admin("This command requires admin privileges.")
-@sopel.module.commands('set')
-@sopel.module.example('.set core.owner Me')
-def set_config(bot, trigger):
-    """See and modify values of Sopel's config object.
-
-    Trigger args:
-        arg1 - section and option, in the form "section.option"
-        arg2 - value
-
-    If there is no section, section will default to "core".
-    If value is None, the option will be deleted.
+def parse_section_option_value(bot, trigger):
+    """Parse trigger for set/unset to get relevant config elements.
+
+    Returns:
+        - False, if invalid command format; let command handle sending usage.
+        - (), if option not found; reply that option is not found.
+        - (section, section_name, static_sec, option, value) otherwise.
+            - value is `None` if omitted from command.
     """
-    # Get section and option from first argument.
     match = trigger.group(3)
     if match is None:
-        bot.reply("Usage: .set section.option value")
-        return
+        return False  # Invalid command
+
+    # Get section and option from first argument.
     arg1 = match.split('.')
     if len(arg1) == 1:
         section_name, option = "core", arg1[0]
     elif len(arg1) == 2:
         section_name, option = arg1
     else:
-        bot.reply("Usage: .set section.option value")
-        return
+        return False  # Invalid command
     section = getattr(bot.config, section_name)
```

Comment:
This will raise `AttributeError` if `bot.config` does not have a section named `section_name`. You should provide a default value here, in order to handle your own control flow.

Reply:
Ah yes, good catch.

Done.